api/src/main/java/org/apache/iceberg/variants/Variant.java

```diff
@@ -25,4 +25,24 @@ public interface Variant {
 
   /** Returns the variant value. */
   VariantValue value();
+
+  /**
+   * Query the value with a given path.
+   *
+   * <p>The path format is a subset of JSON path that supports:
+   *
+   * <ul>
+   *   <li><code>$</code> the root value
+   *   <li><code>.name</code> accesses a field by name
```

Comment:
I'm a little wary of the dot notation because of the collision with dots in field names.  Also, I believe the bracket notation is considered the canonical notation.  I just think we're going to immediately run into an issue with collisions.

Reply:
Bracket notation is canonical, but I want to get something working that we can expand later. I think as long as we define a strict subset we should be able to move on and extend or swap in a library.

Reply:
I'm ok with this as long we keep this internal to the implementation to we can update to support brackets and resolve any pathing limitations with this notation.